### PR TITLE
 Mark config as transient field

### PR DESCRIPTION
**Pull request by hbxie**
Addresses #53.

### CODE REVIEW

File: src/main/java/com/yahoo/bullet/querying/Querier.java [hunk 1]
```diff
@@ -295,7 +296,9 @@
     @Getter(AccessLevel.PACKAGE)
     private RunningQuery runningQuery;
 
-    private BulletConfig config;
+    // Transient field, DO NOT use it beyond constructor and initialize methods.
+    transient private BulletConfig config;
```
Comment (akshaisarma) [inline src/main/java/com/yahoo/bullet/querying/Querier.java]: private transient

File: src/main/java/com/yahoo/bullet/aggregations/SketchingStrategy.java [hunk 2]
```diff
@@ -54,12 +54,12 @@
      */
     @SuppressWarnings("unchecked")
     public SketchingStrategy(Aggregation aggregation, BulletConfig config) {
-        this.config = config;
         metadataKeys = (Map<String, String>) config.getAs(BulletConfig.RESULT_METADATA_METRICS, Map.class);
         separator = config.getAs(BulletConfig.AGGREGATION_COMPOSITE_FIELD_SEPARATOR, String.class);
 
         fieldsToNames = aggregation.getFields();
         fields = Utilities.isEmpty(fieldsToNames) ? Collections.emptyList() : new ArrayList<>(fieldsToNames.keySet());
+        shouldMeta = config.getAs(BulletConfig.RESULT_METADATA_ENABLE, Boolean.class);
```
Comment (akshaisarma) [inline src/main/java/com/yahoo/bullet/aggregations/SketchingStrategy.java]: Move to after 58?

File: src/main/java/com/yahoo/bullet/querying/Querier.java [hunk 3]
```diff
@@ -366,6 +374,10 @@ public Querier(Mode mode, RunningQuery query, BulletConfig config) {
     @Override
     @SuppressWarnings("unchecked")
     public Optional<List<BulletError>> initialize() {
+        int maxCount = config.getAs(BulletConfig.RATE_LIMIT_MAX_EMIT_COUNT, Integer.class);
+        int interval = config.getAs(BulletConfig.RATE_LIMIT_TIME_INTERVAL, Integer.class);
+        maxRateLimit = (maxCount / (double) interval) * RateLimiter.SECOND;
```
Comment (akshaisarma) [inline src/main/java/com/yahoo/bullet/querying/Querier.java]: This is already created and initialized below in the if branch. Should we just make RateLimiter take the config and extract this info and store it?

File: src/main/java/com/yahoo/bullet/aggregations/SketchingStrategy.java (hunk 2 above)
Comment (hbxie) [reply src/main/java/com/yahoo/bullet/aggregations/SketchingStrategy.java]: Done.

File: src/main/java/com/yahoo/bullet/querying/Querier.java (hunk 1 above)
Comment (hbxie) [reply src/main/java/com/yahoo/bullet/querying/Querier.java]: Done.

File: src/main/java/com/yahoo/bullet/querying/Querier.java (hunk 3 above)
Comment (hbxie) [reply src/main/java/com/yahoo/bullet/querying/Querier.java]: Done. Use absoluteRateLimit in RateLimiter instead.

**Pull request merged**